src/ra_server.erl

```diff
@@ -3314,6 +3443,186 @@ after_log_append_reply(Cmd, Idx, Term, Effects0) ->
             Effects0
     end.
 
+post_election_effects(
+  #{cfg := #cfg{effective_machine_version = EffectiveMacVer,
+                machine = Mac}} = State) ->
+    Peers = peers(State),
+    PeerIds = maps:keys(Peers),
+    case PeerIds of
+        [] ->
+            %% This node is alone in the cluster, we can send the `noop'
+            %% command with the newer machine version right away.
+            MacVer = ra_machine:version(Mac),
+            Noop = {noop, #{ts => erlang:system_time(millisecond)},
+                    MacVer},
+            NoopEffect = {next_event, cast, {command, Noop}},
+            [NoopEffect];
+        _ ->
+            %% We continue to send the `noop' command immediately after
```

Comment:
I am wondering if we should build in a mechanism to keep the old behaviour

Reply:
Like an option in the Ra server config to perform the machine upgrade as soon as a leader with the new version is elected as before?

Reply:
yes - just in case someone isn't ready to change to the new behaviour yet or we want to switch back for QQ for example. I'm thinking we could make it a ra system config key rather than a server config key.

Reply:
Thank, let me change that! Any preferences for the name? I’m not satisfied with the `machine_upgrade_when` I came up with.

Reply:
So I moved the option from the server to the system. The name is still `upgrade_machine_when` with two possible values: `after_election` or `when_available_everywhere`, the latter being the default.

That default is set in `src/ra_server.hrl`. `ra_system` queries the application environment and falls back to the default as part of the default system configuration.